### DISCUSSION TITLE
Incremental Save

### DISCUSSION THREAD

**Discussion by pulse-mind** [py-pdf/pypdf#1443] [Q&A] [answered]
Hi

Thanks for the job done on this library. 

Is it possible to do incremental save using pypdf2. Which is required when adding annotation on a pdf that contains a signature.

Best regards,
Fred

**Comment by pubpub-zz** [comment, marked as answer]
Currently the PdfWriter class is working with only one generation of objects. When you append a document or pages, those are copied/imported, losing there original reference. this is not compatible with the incremental update. you may add it the #1181 thread.

**Reply by pulse-mind** [reply]
Thank you for your answer
